src/core/genfb.h

```diff
@@ -156,24 +145,77 @@ class CGenFunctionBlock : public T {
     static size_t getDataPointSpecSize(const CIEC_ANY &paValue);
     static void fillDataPointSpec(const CIEC_ANY &paValue, CStringDictionary::TStringId *&paDataTypeIds);
 
-    static size_t calculateFBConnDataSize(const SFBInterfaceSpec &paInterfaceSpec);
+    virtual void setupFBInterface();
 
-    static size_t calculateFBVarsDataSize(const SFBInterfaceSpec &paInterfaceSpec);
+    virtual void createGenInputData() {
+      // TODO turn into pure virtual as soon as all gen fbs are converted
+    }
 
-    void setupFBInterface();
+    virtual void createGenOutputData() {
+      // TODO turn into pure virtual as soon as all gen fbs are converted
+    }
 
-    void freeFBInterfaceData();
+    void createGenEOCons() {
+      if (getGenEONums()) {
+        mGenEOConns.reserve(getGenEONums());
+        for (size_t i = getGenEOOffset(); i < getGenInterfaceSpec().mNumEOs; ++i) {
+          mGenEOConns.emplace_back(*this, i);
+        }
+      }
+    }
+
+    void createGenDICons() {
+      if (getGenDINums()) {
+        mGenDIConns = std::unique_ptr<CDataConnection *[]>(new CDataConnection *[getGenDINums()]);
+        for (std::size_t i = 0; i < getGenDINums(); ++i) {
+          mGenDIConns[i] = nullptr;
+        }
```

Comment:
I would suggest using the array overload of `std::make_unique`, which also performs value initialization[^1]:
```suggestion
        mGenDIConns = std::make_unique<CDataConnection *[]>(getGenDINums());
```

[^1]: section 2 of https://en.cppreference.com/w/cpp/memory/unique_ptr/make_unique